content/blog/browser-nativefs/index.md

```diff
@@ -0,0 +1,364 @@
+---
+title: Reading and writing files and directories with the browser-nativefs library
+date: 2020-12-08
+note: 'This post appeared first on <a href="https://web.dev/browser-nativefs/">web.dev</a>.'
+author: tomayac
+link: https://twitter.com/tomayac
+---
+
+Browsers have been able to deal with files and directories for a long time.
+The [File API](https://w3c.github.io/FileAPI/)
+provides features for representing file objects in web applications,
+as well as programmatically selecting them and accessing their data.
+The moment you look closer, though, all that glitters is not gold.
+
+<!-- end -->
+
+## The traditional way of dealing with files
+
+<blockquote>
+  If you know how it used to work the old way, you can
+  <a href="#the-file-system-access-api">jump down straight to the new way</a>.
+</blockquote>
+
+### Opening files
+
+As a developer, you can open and read files via the
+[`<input type="file">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file)
+element.
+In its simplest form, opening a file can look something like the code sample below.
+The `input` object gives you a [`FileList`](https://developer.mozilla.org/en-US/docs/Web/API/FileList),
+which in the case below consists of just one
+[`File`](https://developer.mozilla.org/en-US/docs/Web/API/File).
+A `File` is a specific kind of [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob),
+and can be used in any context that a Blob can.
+
+```js
+const openFile = async () => {
+  return new Promise((resolve) => {
+    const input = document.createElement("input");
+    input.type = "file";
+    input.addEventListener("change", () => {
+      resolve(input.files[0]);
+    });
+    input.click();
+  });
+};
+```
+
+### Opening directories
+
+For opening folders (or directories), you can set the
+[`<input webkitdirectory>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-webkitdirectory)
+attribute.
+Apart from that, everything else works the same as above.
+Despite its vendor-prefixed name,
+`webkitdirectory` is not only usable in Chromium and WebKit browsers, but also in the legacy EdgeHTML-based Edge as well as in Firefox.
+
+### Saving (rather: downloading) files
+
+For saving a file, traditionally, you are limited to _downloading_ a file,
+which works thanks to the
+[`<a download>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download:~:text=download)
+attribute.
+Given a Blob, you can set the anchor's `href` attribute to a `blob:` URL that you can get from the
+[`URL.createObjectURL()`](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL)
+method.
+{% Aside 'caution' %}
+To prevent memory leaks, always revoke the URL after the download.
+{% endAside %}
```

Comment:
<img width="706" alt="Screenshot 2020-12-08 at 7 52 27 PM" src="https://user-images.githubusercontent.com/125676/101521751-e8df0000-398e-11eb-9e2f-60f3146971ed.png">

Looks a bit strange.. what is `Aside 'caution'`?

Reply:
Oops, fixed.